lib/openapi_first/request_validation/request_body_validator.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module OpenapiFirst
+  module RequestValidation
+    class RequestBodyValidator
+      def initialize(operation)
+        @operation = operation
+      end
+
+      def validate!(parsed_request_body, request_content_type)
+        request_body = operation.request_body
+        return unless request_body
+
+        schema = request_body.schema_for(request_content_type)
+        unless schema
+          RequestValidation.fail!(:unsupported_media_type,
+                                  message: "#{request_content_type.inspect} is not defined.")
+        end
+
+        RequestValidation.fail!(:invalid_body) if request_body.required? && parsed_request_body.nil?
```

Comment:
this results in an error with the message "Request body invalid: ". I'd go with something more descriptive such as:
```suggestion
        RequestValidation.fail!(:invalid_body, message: "request body is not defined.") if request_body.required? && parsed_request_body.nil?
```